### ISSUE TITLE
Return type of mos_getkbmap()

### ISSUE THREAD

**Issue by avysk** [pcawte/AgDev#48]
Hello,

the return type of `mos_getkbmap` is declared as `uint8_t *` (see https://github.com/pcawte/AgDev/blob/cc999444d435ecf975dc2996ade0bbef68ae82d9/src/libc/include/mos_api.h#L293).

However, MOS documentation (https://agonconsole8.github.io/agon-docs/MOS-API/#0x1e-mos_getkbmap) says the return value is 24-bit.

**Comment by HeathenUK** [comment]
This is correct. It's a 24-bit address pointing to a 32-byte (from memory, could be less?) memory area.

There's no wrong or right way to cast it and `uint8_t` is probably the most sensible default. Again, you can cast it to whatever you like if you want to access it as 16 x `uint16_t` or 8 x `uint32_t` instead of 32 x `uint8_t`.

**Comment by avysk** [comment]
Ah, stupid me. I misread the header. Sorry.
